### ISSUE TITLE
read_raw_edf fails if excluding all the signal channels

### ISSUE THREAD

**Issue by laemtl** [mne-tools/mne-python#9811]
#### Describe the bug
When calling read_raw_edf(path, exclude=exclude, preload=False), if exclude contains all channels but stim ones, 
the call fails 

```
  File "/data/loris-mri/bin/mri/python_virtualenvs/loris-mri-python/lib/python3.7/site-packages/mne/io/edf/edf.py", line 433, in _get_info
    edf_info['record_length'][1] / edf_info['record_length'][0]
  File "/data/loris-mri/bin/mri/python_virtualenvs/loris-mri-python/lib/python3.7/site-packages/numpy/core/_methods.py", line 40, in _amax
    return umr_maximum(a, axis, None, out, keepdims, initial, where)
ValueError: zero-size array to reduction operation maximum which has no identity

```
In mne/io/edf/edf.py::_get_info
```
sfreq = np.take(n_samps, not_stim_ch).max() * \
        edf_info['record_length'][1] / edf_info['record_length'][0]

```
np.take(n_samps, not_stim_ch) returns an empty list and .max() is failing.

#### Steps to reproduce
Calling read_raw_edf(path, exclude=exclude, preload=False), with exclude set to all channels but stim ones 

#### Expected results
Avoid the ValueError and report the absence of data/signal channels to the user

#### Actual results
ValueError: zero-size array to reduction operation maximum which has no identity

**Comment by laemtl** [comment]
Sure: 
Click on login here: https://demo.loris.ca
Then you can download a sample using this link: https://demo.loris.ca/mri/jiv/get_file.php?file=bids_imports/Face13_BIDSVersion_1.1.0/sub-OTT170/ses-V1/eeg/sub-OTT170_ses-V1_task-faceO_eeg.edf

**Comment by laemtl** [comment]
The file is not problematic. The problem occurs when we try to read a file with all channels ignored but stim ones.
You will get the user is not authenticated error if you don't follow the first step: 
- Click on the login button here: https://demo.loris.ca
Once this is done, you can download the file.

Hope this helps!

**Comment by agramfort** [comment]
ok I could replicate with:

import mne

path = "/Users/alex/Downloads/sub-OTT170_ses-V1_task-faceO_eeg.edf"
exclude = ()
raw = mne.io.read_raw_edf(path, exclude=exclude, preload=False)
exclude = raw.ch_names[:-1]
raw = mne.io.read_raw_edf(path, exclude=exclude, preload=False)

**Comment by cbrnr** [comment]
We do not support loading EDF files without any data channels. We've fixed that during loading in #9283, but apparently we haven't considered this use case.
